sympy/core/relational.py

```diff
@@ -180,13 +206,24 @@ def canonical(self):
         """
         args = self.args
         r = self
-        if r.rhs.is_Number:
-            if r.lhs.is_Number and r.lhs > r.rhs:
+        if r.rhs.is_number:
+            if r.rhs.is_Number and r.lhs.is_Number and r.lhs > r.rhs:
                 r = r.reversed
-        elif r.lhs.is_Number:
+        elif r.lhs.is_number:
             r = r.reversed
         elif tuple(ordered(args)) != args:
             r = r.reversed
+
+        # Check if first value has negative sign
+        if not isinstance(r.lhs, BooleanAtom) and r.lhs.could_extract_minus_sign():
+            r = r.reversedsign
+        elif not isinstance(r.rhs, BooleanAtom) and not r.rhs.is_number and r.rhs.could_extract_minus_sign():
+            # Right hand side have a minus, but not lhs.
```

Comment:
Should say `has`, but probably not worth a new test run. Will try to remember it for a later PR.